torch/jit/_recursive.py

```diff
@@ -147,13 +151,16 @@ def infer_type(name, item):
             concrete_type_builder.add_attribute(name, attr_type, False, False)
             continue
         if attr_type is not None:
-            assert attr_type.is_interface_type()
-            # if the type can be inferred, it should be a module interface type
-            sub_concrete_type = torch._C.ConcreteModuleType.from_jit_type(attr_type)
+            if attr_type.is_interface_type():
```

Comment:
What's going on here exactly ? i'm not following

Reply:
So as far as I know, at the moment, if a submodule is a key in `__annotations__`, it can only map to a module interface type. This PR changes that (it can map to a contained type hint also), so we need to handle both cases and can no longer be certain that it must be a module interface type.